core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/LimitedSizeLineConsumerWrapper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.core.util;
+
+import java.io.IOException;
+
+/**
+ * Wrapper of {@link LineConsumer} which limits total size of consuming lines.
+ *
+ * @author Dmytro Nochevnov
+ * @author Anatolii Bazko
+ */
+public class LimitedSizeLineConsumerWrapper implements LineConsumer {
+  private final int limit;
+  private final LineConsumer lineConsumer;
+  private int totalLength;
+
+  /**
+   * @param limit maximum total size of consuming lines.
+   * @throws IllegalArgumentException if the {@code limit} is negative.
+   */
+  public LimitedSizeLineConsumerWrapper(LineConsumer lineConsumer, int limit) {
```

Comment:
If we have `LimitedSize` class prefix then the parameter should be `size` but not `limit`

Reply:
Frankly speaking, as for me, class **LimitedLineConsumerWrapper** with **limit** field inside looks more obvious, similar to the class [LimitedInputStream](https://github.com/eclipse/che/blob/2c2bd6637c8169018e2b33397f9d738bbf82a0b1/infrastructures/docker/docker-client/src/main/java/org/eclipse/che/infrastructure/docker/client/connection/LimitedInputStream.java#L17).